proto/backend.proto

```diff
@@ -131,7 +131,22 @@ message DataResponse {
   // Frame has its own meta, warnings, and repeats refId
   repeated bytes frames = 1;
   string error = 2;
-  bytes jsonMeta = 3; // Warning: Current ignored by frontend. Would be for metadata about the query.
+  bytes jsonMeta = 3; // Warning: Currently ignored by frontend. Would be for metadata about the query.
+  Status status = 4;
+
+  enum Status {
```

Comment:
given that these all map to HTTP status codes... maybe at least hardcode them to the status codes?  Kinda sucks for the non-go users that will need to reverse engineer this from: https://github.com/grafana/grafana-plugin-sdk-go/blob/0f693ea3a43f1f5d405faf7cec6f1d8e6667416b/backend/error_status.go#L117

Reply:
Non-go users don't have to care about HTTP statuses because it's only rendered in the JSON representation of QueryDataResponse which is happening in Grafana (with the help from the Go SDK). When compiling code of protobuf for other languages you'll operate on those statuses rather than HTTP statuses.

At least right now I don't see the need for HTTP statuses for non-go users, but maybe missing something/not thinking straight about it?

Reply:
> At least right now I don't see the need for HTTP statuses for non-go users

My problem with it just that it masks what is actually happening and making an unnecessary distinction in the go-sdk from the transport layer (json or protobuf)

The need in "non-go" is identical to the need for go, why introduce special cases and surprises that must be reasoned about?   Also, we do we want or need to limit the codes?  why not 412?  or whatever response code is actually returned from the query (avoiding the need to then map it to a new response code)!

Reply:
This is not a huge deal either way -- just trying to understand what we gain by not trusting developers to put the codes they are most familiar with (and likely +90% of the time will be mapping from an exact HTTP status code on the server!)

Reply:
Thinking on this some more... likely best to just ask the enterprise plugins team how they would want to work with it.